### PR TITLE
[4.0] Module position no more empty when coming from CPanel

### PR DESCRIPTION
**Pull request by infograf768**
Pull Request for Issue #24575

### Summary of Changes
Forcing userstate to administrator modules

### Testing Instructions
Load Site Modules Manager to be sure userstate is set to site modules
Switch to Control Panel
Edit a module through the cog

### Before patch
The module position is empty

### After patch
It is now correctly using cpanel position

@chmst @franz-wohlkoenig

### CODE REVIEW

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on a0846e416653b354033415c787db0f1cdd063997<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24583">issues.joomla.org/tracker/joomla-cms/24583</a>.</sub>

**Comment by infograf768** [comment]
Solved conflicts.
One more tester.

File: administrator/templates/atum/html/layouts/chromes/well.php [hunk 1]
```diff
@@ -39,6 +39,8 @@
 	<div class="<?php echo $moduleClass; ?> module-wrapper">
 		<<?php echo $moduleTag; ?> class="card mb-3<?php echo $moduleClassSfx; ?>">
 			<?php if ($canEdit || $canChange) : ?>
+				<?php // Make sure we get the admin template module positions ?>
+				<?php Factory::getApplication()->setUserState('com_modules.modules.client_id', 1, 'int'); ?>
```
Comment (Quy) [inline administrator/templates/atum/html/layouts/chromes/well.php]: Is this necessary `'int'`?

Comment (infograf768) [reply administrator/templates/atum/html/layouts/chromes/well.php]: not sure

Comment (Quy) [reply administrator/templates/atum/html/layouts/chromes/well.php]: Checked similar calls and they don't include `'int'`.

Comment (infograf768) [reply administrator/templates/atum/html/layouts/chromes/well.php]: done

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 286b10cff2def7338673718bee64f9a2b249664b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24583">issues.joomla.org/tracker/joomla-cms/24583</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by SharkyKZ** [comment]
This is not the right approach, IMO. The layout should not be setting user state. This way admin filter is forced whenever a page containing a module with this layout is visited.

**Comment by infograf768** [comment]
@SharkyKZ 
There are 2 possible solutions:
1. Check if the position is `cpanel` and set userstate in the layout.
This would work for all admin modules in the cpanel position and only for these

2. The only other solution I see is to add in EACH admin module helper the code
`Factory::getApplication()->setUserState('com_modules.modules.client_id', 1);`

What do you propose?

**Comment by SharkyKZ** [comment]
One option is to get `client_id` from form data in the field here:

https://github.com/joomla/joomla-cms/blob/5f7bb276943a3a1dc89f3eebf670fbfbb2d29028/administrator/components/com_modules/Field/ModulesPositioneditField.php#L52

```
$clientId  = $this->form->getValue('client_id', '', Factory::getApplication()->getUserState('com_modules.modules.client_id', 0));
```
Other option would be to add `client_id` to URL and have controller set the user state based on that.

**Comment by infograf768** [comment]
Please make the PR yourself as I do not understand what you propose.

**Comment by SharkyKZ** [comment]
See PR #24698 for alternative.

**Comment by infograf768** [comment]
Closing in favour of https://github.com/joomla/joomla-cms/pull/24698
Thanks @SharkyKZ

**Pull request closed without merging**